source/slang/slang-ir-lower-combined-texture-sampler.cpp

```diff
@@ -0,0 +1,208 @@
+                                        descriptorTableSlotOffsetAttr = offsetAttr;
+                                    auto info = newVarLayoutBuilder.findOrAddResourceInfo(offsetAttr->getResourceKind());
+                                    info->offset = offsetAttr->getOffset();
+                                    info->space = offsetAttr->getSpace();
+                                    info->kind = offsetAttr->getResourceKind();
+                                }
+                                // If the user provided an layout offset for the texture but not for descriptor table slot, then
+                                // we use the texture offset for the descriptor table slot offset.
+                                if (resOffsetAttr && !descriptorTableSlotOffsetAttr)
+                                {
+                                    auto info = newVarLayoutBuilder.findOrAddResourceInfo(LayoutResourceKind::DescriptorTableSlot);
+                                    info->offset = resOffsetAttr->getOffset();
+                                    info->space = resOffsetAttr->getSpace();
+                                    info->kind = LayoutResourceKind::DescriptorTableSlot;
+                                }
+                                auto newVarLayout = newVarLayoutBuilder.build();
+                                subBuilder.addLayoutDecoration(typeUser, newVarLayout);
+                                varLayout->removeAndDeallocate();
+                            }
+                        }
+                    }
+                }
+            }
+        }
+
+        // If no combined texture sampler type exist in the IR module,
+        // we can exit now.
+        if (context.mapTypeToLoweredInfo.getCount() == 0)
+            return;
+
+        // We need to process all insts in the module, and replace
+        // CombinedTextureSamplerGetTexture and CombinedTextureSamplerGetSampler into
+        // FieldExtracts.
+        IRBuilder builder(module);
+        for (auto globalInst : module->getGlobalInsts())
+        {
+            auto func = as<IRFunc>(getGenericReturnVal(globalInst));
+            if (!func)
+                continue;
+            for (auto block : func->getBlocks())
+            {
+                IRInst* nextInst = nullptr;
+                for (auto inst = block->getFirstInst(); inst; inst = nextInst)
+                {
+                    nextInst = inst->getNextInst();
+                    switch (inst->getOp())
+                    {
+                    case kIROp_CombinedTextureSamplerGetTexture:
+                    case kIROp_CombinedTextureSamplerGetSampler:
+                        {
+                            auto combinedSamplerType = inst->getOperand(0)->getDataType();
+                            auto loweredInfo = context.mapTypeToLoweredInfo.tryGetValue(combinedSamplerType);
+                            if (!loweredInfo) continue;
```

Comment:
```suggestion
                            if (!loweredInfo) 
                                continue;
```
new line continue to allow debugging (code step-ing)